--- sphinx-0.9.7-rc2/src/sphinx.cpp~	2006-12-14 23:21:06.000000000 +0200
+++ sphinx-0.9.7-rc2/src/sphinx.cpp	2007-03-13 10:49:26.646111169 +0200
@@ -87,7 +87,7 @@
 // COMPILE-TIME CHECKS
 /////////////////////////////////////////////////////////////////////////////
 
-STATIC_SIZE_ASSERT ( SphOffset_t, 8 );
+STATIC_SIZE_ASSERT ( SphOffset_t, 4 );
 
 /////////////////////////////////////////////////////////////////////////////
 // INTERNAL PROFILER
